src/test/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/AwsDatasetModificationIT.kt

```diff
@@ -1,30 +1,30 @@
-package com.atlassian.performance.tools.awsinfrastructure
+package com.atlassian.performance.tools.awsinfrastructure.api
 
 import com.amazonaws.regions.Regions
 import com.atlassian.performance.tools.aws.api.StorageLocation
 import com.atlassian.performance.tools.awsinfrastructure.IntegrationTestRuntime.aws
 import com.atlassian.performance.tools.awsinfrastructure.IntegrationTestRuntime.taskWorkspace
-import com.atlassian.performance.tools.awsinfrastructure.api.DatasetCatalogue
 import com.atlassian.performance.tools.infrastructure.api.dataset.Dataset
 import com.atlassian.performance.tools.ssh.api.Ssh
 import org.junit.Test
 import java.net.URI
 import java.time.Duration.ofMinutes
-import java.util.*
 
-class AwsDatasetIT {
+class AwsDatasetModificationIT {
 
     private val workspace = taskWorkspace.isolateTest(javaClass.simpleName)
     private val sourceDataset = smallJiraSeven()
 
     @Test
     fun shouldRemoveBackups() {
-        AwsDataset(sourceDataset)
-            .modify(
-                aws = aws,
-                workspace = workspace,
-                newDatasetName = "dataset-${UUID.randomUUID()}"
-            ) { infrastructure ->
+        val modification = AwsDatasetModification.Builder(
+            aws = aws,
+            dataset = sourceDataset
+        )
+            .workspace(workspace)
+            .build()
+        val transformation = object : DatasetTransformation {
```

Comment:
If we go back to `Consumer<Infrastructure<*>`, then we can:
```kotlin
modification.apply { infrastructure ->
    val jiraHome = infrastructure.jira.jiraHome
    val backupPath = "${jiraHome.location}/export"
    Ssh(jiraHome.host, connectivityPatience = 4)
        .newConnection()
        .use { ssh ->
            val listCommand = "ls -lh $backupPath"
            val listOutput = ssh.execute(listCommand).output
            println("$ $listCommand\n$listOutput")
            ssh.execute("rm -r $backupPath")
        }
}
```

Reply:
How it differs for `DatasetTransformation`?

Reply:
Do you mean it's more readable with `Consumer<Infrastructure<*>`?

Reply:
It's more readable, because you don't need anonymous implementations, just lambdas.